src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultRazorConfigurationService.cs

```diff
@@ -83,7 +93,19 @@ private RazorLSPOptions BuildOptions(IConfiguration config)
                 enableFormatting = parsedEnableFormatting;
             }
 
-            return new RazorLSPOptions(trace, enableFormatting);
+            var tabSize = instance.TabSize;
+            if (int.TryParse(config["tabSize"], out var parsedTabSize))
```

Comment:
Is there a guarantee that `config` will always have a `tabSize`?

Reply:
Guarantee is a strong word but we can expect vscode to do the right thing. Moreover, if it is not there it still won't throw. The default implementation of `IConfiguration` will just return null.